datafusion/sqllogictest/test_files/map.slt

```diff
@@ -493,3 +509,68 @@ select cardinality(map([1, 2, 3], ['a', 'b', 'c'])), cardinality(MAP {'a': 1, 'b
        cardinality(MAP {'a': MAP {1:'a', 2:'b', 3:'c'}, 'b': MAP {2:'c', 4:'d'} });
 ----
 3 2 0 2
+
+# map_extract
+# key is string
+query IIII
+select map_extract(MAP {'a': 1, 'b': NULL, 'c': 3}, 'a'), map_extract(MAP {'a': 1, 'b': NULL, 'c': 3}, 'b'),
+       map_extract(MAP {'a': 1, 'b': NULL, 'c': 3}, 'c'), map_extract(MAP {'a': 1, 'b': NULL, 'c': 3}, 'd');
+----
+1 NULL 3 NULL
```

Comment:
By my understanding of `duckdb`'s doc the values should be a list not a scalar. In other words the result should be  `[1], [], [3], []` not `1, NULL, 3, NULL`:

```sql
D select map_extract(MAP {'a': 1, 'b': NULL, 'c': 3}, 'a');
┌─────────────────────────────────────────────────────────────────────────────────────────┐
│ map_extract(main.map(main.list_value('a', 'b', 'c'), main.list_value(1, NULL, 3)), 'a') │
│                                         int32[]                                         │
├─────────────────────────────────────────────────────────────────────────────────────────┤
│ [1]                                                                                     │
└─────────────────────────────────────────────────────────────────────────────────────────┘
D
D select map_extract(MAP {'a': 1, 'b': NULL, 'c': 3}, 'd');
┌─────────────────────────────────────────────────────────────────────────────────────────┐
│ map_extract(main.map(main.list_value('a', 'b', 'c'), main.list_value(1, NULL, 3)), 'd') │
│                                         int32[]                                         │
├─────────────────────────────────────────────────────────────────────────────────────────┤
│ []                                                                                      │
└─────────────────────────────────────────────────────────────────────────────────────────┘
```

Reply:
WHile this choice is strange to me (a list that will always have 1 or 0 elements) I think it will make for an easy implementation -- you shouldn't have to make special implementations for each type of key -- instead you should just be able to calculate the offsets directly and make a list array that points to the keys array

Reply:
@Weijun-H Here is a like to old converstation about how DataFuion strives not to invent / determine semantics.
https://github.com/apache/datafusion/issues/11436#issuecomment-2275635437

Reply:
I find it unintuitive to return a list in this context because, in a Map, each key is unique by definition. Returning a list implies the possibility of multiple values for a single key, which contradicts the core concept of how a Map operates. The key uniqueness in a Map should naturally lead to a single value or `NULL`, making a list redundant and potentially confusing. I refactored the code so it can avoid to make special implementations for each type of key now.

Reply:
I check the history of map_extract in duckdb and find that there is no discussion about it. It is created by one man in early day
https://github.com/duckdb/duckdb/blob/a196a022ffcf427aa21184d295bb70caa42e0655/src/function/scalar/nested/map_extract.cpp

Given the rationale is not clear, I don't have strong opinion to follow what duckdb's result is.

Reply:
> It is created by one man 

I think this describes quite a bit of duckdb 😆